core/src/main/java/org/mule/runtime/core/processor/strategy/NonBlockingProcessingStrategyFactory.java

```diff
@@ -168,20 +60,22 @@ public void configureProcessors(List<Processor> processors,
     public Function<Publisher<Event>, Publisher<Event>> onProcessor(Processor processor,
                                                                     Function<Publisher<Event>, Publisher<Event>> publisherFunction) {
       if (processor instanceof NonBlockingMessageProcessor) {
-        return publisher -> from(publisher).transform(publisherFunction).publishOn(fromExecutorService(executorService));
+        return publisher -> from(publisher).transform(publisherFunction).publishOn(fromExecutorService(scheduler));
       } else {
-        return super.onProcessor(processor, publisherFunction);
+        return publisher -> from(publisher).transform(publisherFunction);
       }
     }
 
     @Override
     public void start() throws MuleException {
-      executorService = newFixedThreadPool(maxThreads);
+      this.scheduler = schedulerService.cpuLightScheduler();
```

Comment:
This should IO pool, as blocking operations occur using these thread. (Today pool of 128 is used by default in 3.x)